integration-cli/docker_cli_build_test.go

```diff
@@ -3378,42 +2555,40 @@ func (s *DockerSuite) TestBuildDockerignoreTouchDockerfile(c *check.C) {
 	if id1 != id2 {
 		c.Fatalf("Didn't use the cache - 3")
 	}
-
 }
 
 func (s *DockerSuite) TestBuildDockerignoringWholeDir(c *check.C) {
```

Comment:
This test was really weird. The last 3 builds were wrong because a change in `.dockerignore` doesn't seem to invalidate the cache. If it did, this tests should have been red as `.*` ignores only *dotfiles*, `*` ignore all and `?`, `.` ignore pretty much nothing.

/cc @duglin @tiborvass @LK4D4